[Wong Wei Han] iP

WeiHanWong opened the pull request:
# Will

> "Your mind is for having ideas, not holding them." – David Allen [(source)](https://en.wikipedia.org/wiki/David_Allen_(productivity_consultant))

Will frees your mind of having to remember things you need to do. It's,

- **text-based**
- **easy to learn**
- **FAST SUPER FAST** to use

All you need to do is,

1. download it from [here](https://example.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😏

And it is **FREE**!

### Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

WeiHanWong committed:
Level 0. Rename, Greet, Exit

WeiHanWong committed:
Level 1. Echo

WeiHanWong committed:
Level 2. Add, List

WeiHanWong committed:
Level 3. Mark as Done

Commit:
Level 4. ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Level 5. Handle Errors

Commit:
Level 6. Delete

Commit:
Use Enums

Commit:
Add support for persistent memory

Commit:
Add date and time recognition function

Commit:
Make the code more oop

Commit:
Organise into Packages

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
Add JUnit Tests

Commit:
Add JavaDoc comments

Commit:
Follow the Coding Standard

Commit:
Add find function

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Follow the Coding Standard

# Conflicts:
#	src/main/java/Utilities/Ui.java

Commit:
Resolve Merge conflict

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find function

# Conflicts:
#	src/main/java/Utilities/Ui.java

Commit:
Use checkStyle to detect coding style violations

Commit:
Add a GUI to the chatbot

Commit:
Use Java varargs feature

Commit:
Add css to javafx

dez2003 commented on src/main/java/Utilities/Storage.java:
```diff
@@ -0,0 +1,107 @@
+import Task.Task;
+import Task.ToDo;
+
+import java.io.*;
```
Following the coding standard importing classes should be done explicitly

dez2003 commented on src/main/java/Utilities/Storage.java:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    /**
+     * Format task object to string.
```
```suggestion
     * Formats task object to string.
```

dez2003 commented on src/main/java/Utilities/Parser.java:
```diff
@@ -0,0 +1,286 @@
+                    return output + result;
+                }
+            }catch(StringIndexOutOfBoundsException e){
+                return "The keyword to look for cannot be empty.";
```
Make sure to add white spacing in your statements!

marcusjhang commented on src/main/java/Frameworks/Will.java:
```diff
@@ -0,0 +1,69 @@
+package Frameworks;
```
package name should be in lowercase -> frameworks

marcusjhang commented on src/main/java/Frameworks/Will.java:
```diff
@@ -0,0 +1,69 @@
+
+        TaskList taskList = new TaskList();
+        Parser dateTimeParser = new Parser();
+        Parser parser = new Parser();
```
The variable dateTimeParser and parser might be confusing because both are instances of Parser. Use distinct names that reflect their roles, such as commandParser and dateParser.

marcusjhang commented on src/main/java/Utilities/Ui.java:
```diff
@@ -0,0 +1,148 @@
+     *
+     * @param logo The name or logo of the application.
+     */
+    public void greetMsg(String logo) {
```
greetMsg does not follow the naming convention of being descriptive and clear. Rename greetMsg to something like displayGreetingMessage.

marcusjhang replied:
Use standard words and avoid abbreviations unless they are well understood. Replace Msg with Message

Commit:
Improve coding standards

dez2003 commented on src/main/java/Utilities/Parser.java:
```diff
@@ -0,0 +1,286 @@
+     * @param taskList       The TaskList object to manage tasks.
+     * @param dateTimeParser The Parser object used to parse date and time strings.
+     * @param storage        The Storage object used to save user tasks.
+     * @return A string representing the result of the executed command.
```
Nice, I like how you lined the param descriptions up so it's easier to read!

dez2003 reviewed (commented):
Overall, good job with abiding by the coding standards! Just a few minor tweaks here and there

Commit:
Add assertion in the code

The code does not verify certain assumptions, leading to potential undetected issues during runtime.

This change ensures that critical conditions are validated.

Use assertions to prevent invalid data and catch potential bugs early in development.

Assertions are lightweight, and they help identify logic errors in a controlled environment without affecting performance in production.

The assertions will help maintain code integrity and improve debugging efficiency.

Commit:
Improve code quality

Commit:
Use Streams

Commit:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertion in the code

Commit:
Merge branch 'master' into branch-A-Streams

* master:
  Add assertion in the code

Commit:
Merge branch 'master' into branch-A-Streams

* master:
  Improve code quality

Commit:
Set up CI

WeiHanWong committed:
Update gradle.yml

Commit:
Add the ability to recognise and deal with duplicate items

WeiHanWong committed:
Merge pull request #1 from WeiHanWong/branch-C-DetectDuplicates

Add the ability to recognise and deal with duplicate items

Commit:
Add a User Guide and Ui

Commit:
Fix bugs from test

Commit:
Add more javadocs

Commit:
Improve the GUI

Commit:
Improve the code to handle more errors

Commit:
Add more tests

Commit:
Prepare for build

WeiHanWong committed:
Improve Code Standard

Commit:
Fix code quality
